tests/lib/prepare.sh

```diff
@@ -145,5 +148,9 @@ prepare_all_snap() {
         fi
     fi
 
+    # ensure no test creates havoc
+    for name in pc pc-kernel ubuntu-core; do
+        snap list | grep $name
```

Comment:
Yeah that explains the goal of the test (which is an improvement), but not why the test failed. Leaving it like that actually makes it look like the `snap list` command failed, which isn't the case. Since my PR related to the socket, I initially started wondering if I messed up the API somehow, when in reality it was the grep that failed. Note that it's not clear from the failure that `snap list` is being piped to grep, either.